core/lib/globals_api.h

```diff
@@ -664,19 +664,25 @@ typedef uint64 dr_opmask_t;
 
 #if defined(AARCHXX)
 /**
- * 128-bit ARM SIMD Vn register.
- * In AArch64, align to 16 bytes for better performance.
- * In AArch32, we're not using any uint64 fields here to avoid alignment
- * padding in sensitive structs. We could alternatively use pragma pack.
+ * 512-bit ARM Scalable Vector Extension (SVE) vector registers Zn and
+ * predicate registers Pn. Low 128 bits of Zn overlap with existing ARM
+ * Advanced SIMD (NEON) Vn registers. The SVE specification defines the
+ * following valid vector lengths:
+ * 128 256 384 512 640 768 896 1024 1152 1280 1408 1536 1664 1792 1920 2048
+ * We currently support 512-bit maximum due to DR's stack size limitation,
+ * (machine context stored in the stack). In AArch64, align to 16 bytes for
+ * better performance. In AArch32, we're not using any uint64 fields here to
+ * avoid alignment padding in sensitive structs. We could alternatively use
+ * pragma pack.
  */
 #    ifdef X64
 typedef union ALIGN_VAR(16) _dr_simd_t {
-    byte b;      /**< Bottom  8 bits of Vn == Bn. */
-    ushort h;    /**< Bottom 16 bits of Vn == Hn. */
-    uint s;      /**< Bottom 32 bits of Vn == Sn. */
-    uint d[2];   /**< Bottom 64 bits of Vn == Dn as d[1]:d[0]. */
-    uint q[4];   /**< 128-bit Qn as q[3]:q[2]:q[1]:q[0]. */
-    uint u32[4]; /**< The full 128-bit register. */
+    byte b;       /**< Bottom  8 bits of Vn, Zn and Pn, (Bn). */
```

Comment:
I don't understand this list syntax: is it grouping Vn and Zn into a pair, and Pn and Bn as another pair: but why is Bn in parens but Zn is not.

Reply:
What those 4 comments are trying to say is that there are 3 vector types in AArch64:
Vn = SIMD vector registers
Zn = SVE vector registers
Pn = SVE predicate registers
`/**< Bottom  8 bits of Vn, Zn and Pn,`

And each vector has a scalar byte element type
`, (Bn). */`

That applies to H, S and D scalar element types as well. Perhaps this is better?
```
/**< Byte scalar element of Vn, Zn and Pn.                */
/**< Halfword (16 bit) scalar element of Vn, Zn and Pn.   */
/**< Singleword (32 bit) scalar element of Vn, Zn and Pn. */
/**< Doubleword (64 bit) scalar element of Vn, Zn and Pn. */
```

Reply:
OK, the suggestion looks good.  Could put Bn in there somehwere: `Byte scaler element (Bn) of Vn, Zn, or Pn.`

Reply:
Done. Updated comment block with better description and more details.